server.js

```diff
@@ -2,38 +2,59 @@
 
 // not using import, using require instead
 // requires/imports express
+require('dotenv').config(); // lets you use env file
+const axios = require('axios');
 const express = require('express');
 const cors = require('cors');
-require('dotenv').config(); // lets you use env file
 const weatherData = require('./data/weather.json');
-
 const app = express(); // gives back return value
-console.log(app);
-
 app.use(cors()); // acts as middleware - intermediary between server and requests
 
+// opens up a route for /weather, calls handleGetWeather when route recieves query
 app.get('/weather', handleGetWeather);
+// send error back to client if page not found
+app.get('/*', (req, res) => res.status(404).send('Route Not Found'));
+// turn on server
+app.listen(process.env.PORT, () => console.log('server is listening on PORT 3001'));
+
 
+async function handleGetWeather(req, res) {
+  // set up endpoint to accept city_name as param
+  console.log(req.query);
+  let city_name = req.query.city_name;
+  console.log(city_name);
+  //let cities_in_data = ['Seattle', 'Paris', 'Ammon'];
+  let city_match = weatherData.find(city => city.city_name.toLowerCase() === city_name.toLowerCase());
+  console.log('city_match', city_match);
+
+  if (city_match) {
+    // map to Forecast, returns obj with datetime and description as properties
+    const resForecast = city_match.data.map(day => new Forecast(day));
+    // send city weather data description to client
+    console.log(resForecast);
+    res.status(200).send(resForecast);
+
+  } else {
+    const liveWeather = await axios.get(`http://api.weatherbit.io/v2.0/forecast/daily?lat=${req.query.lat}&lon=${req.query.lon}&key=${process.env.WEATHER_API_KEY}&units=I`);
+    // makes new property on weather object for searched city
+    weatherData[liveWeather.data.city_name] = liveWeather.data;
+    console.log(liveWeather);
+    res.status(400).send(`${city_name} not found.`);
+  }
```

Comment:
Great work on working through the lab 👍 
 If  using try catch blocks for this `async` function is of interest then we can do this:

- Once you get into lab 8 a bit deeper, we can refactor. The GET request with axios will be in your `try` block. Inside this block we will also send the weather data array to the client.
- Your `catch` block will then catch errors and send a status of `500` should something go wrong 👍